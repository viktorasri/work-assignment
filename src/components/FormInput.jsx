import React from 'react';
import PropTypes from 'prop-types';

const FormInput = (props) => {
  return (
    <div className='form__control'>
      {/* <input className='form__input' onChange={onChange} type={type} placeholder={placeholder} {...otherProps} /> */}
      <input className='form__input' {...props} />
    </div>
  );
};

FormInput.defaultProps = {
  type: 'text',
};

FormInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
};

export default FormInput;
