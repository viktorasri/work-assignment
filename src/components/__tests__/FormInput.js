import React from 'react';
import { shallow } from 'enzyme';
import FormInput from '../FormInput';

describe('<FormInput />', () => {
  const mockOnchange = jest.fn();
  const mockProps = {
    onChange: mockOnchange,
    value: 'test',
    placeholder: 'Form input',
  };

  const wrapper = shallow(<FormInput {...mockProps} />);

  it('expect to render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    wrapper.find('input').simulate('change');
    expect(mockOnchange).toHaveBeenCalled();
  });
});
