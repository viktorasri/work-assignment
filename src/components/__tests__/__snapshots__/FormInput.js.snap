// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> expect to render component 1`] = `
<div
  className="form__control"
>
  <input
    className="form__input"
    onChange={[MockFunction]}
    placeholder="Form input"
    type="text"
    value="test"
  />
</div>
`;
